File: Difficulty: Medium/Sort a stack/sort-a-stack.py
class Solution:
    # your task is to complete this function
    # function sort the stack such that top element is max
    # funciton should return nothing
    # s is a stack
    def Sorted(self, s):
        self.sort(s)
        
        
    def sort(self,s):
        if (len(s)==1):
            return 
        temp = s.pop()
        self.sort(s)
        self.insert(s,temp)
        
        
    def insert(self,s, temp):
        if(len(s)==0 or s[-1]<= temp):
            s.append(temp)
            return
        val = s.pop()
        self.insert(s,temp)
        s.append(val)

File: Difficulty: Medium/Sort a stack/sort-a-stack.cpp
/*The structure of the class is
class SortedStack{
public:
    stack<int> s;
    void sort();
};
*/

/* The below method sorts the stack s
you are required to complete the below method */
void insertStack(stack<int> &s, int x){
    if(s.size()==0 || x>s.top()){
        s.push(x);
        return;
    }
    int temp = s.top();
    s.pop();
    insertStack(s,x);
    s.push(temp);
}
void SortedStack ::sort() {
    // Your code here
    if(s.size() == 0)
        return;
    int temp = s.top();
    s.pop();
    sort();
    insertStack(s, temp);
}